### PR TITLE
Fix incompatibility between different connectors for replace methods

### PR DESCRIPTION
**Pull request by Amir-61**
* The goal is to to achieve the following behaviour for missing values:

db | replaceById / replaceAttributes | replaceOrCreate
--- | --- | ---
*NoDQL (MongoDB)* | undefined| undefined
*NoSQL (Memory)* | undefined   |  undefined
*SQL DBs (Mysql)* |  undefined|   undefined

#### Test statuses: 

dependencies' tests | status
--- | --- | ---
*Loopback-connector-Mysql* | Aborted! [1] 👎 
*Loopback-connector-MSsql* | passed 👍 
*Loopback-connector-Oracle* |  passed 👍 
*Loopback-connector-Postgressql* |  passed 👍 
*Loopback-connector-mongodb* |  passed 👍 
*Loopback-connector-redis* |  passed 👍 
*Loopback-datasource-juggler* |  passed 👍 
*Loopback-connector-sqllite* |  first time failed, but second time passed [2] 👎 

[1]:[Here](http://ci.strongloop.com/job/loopback-connector-mysql/label=ubuntu-0.12/1250/console), The tests for `loopback-connector-mysql` gets aborted and I don't know why! Hey @rmg did you abort the `loopback-connector-mysql` tests?

[2]: [Here](http://ci.strongloop.com/job/loopback-connector-sqlite3/label=amazon-5/572/console) I get this failure: `Deprecation warning: moment construction falls back to js Date. This is discouraged and will be removed in upcoming major release. Please refer to https://github.com/moment/moment/issues/1407 for more info.`; again I don't think if it is relevant to this PR... Also the final result in this repos says passing:
```
[sl-ci-run INFO  234.4 ]: *************** BUILD FRAMEWORK REPORT CARD ****************
[sl-ci-run INFO  234.4 ]: **  Sanity checks         : pass   [required]
[sl-ci-run INFO  234.4 ]: **  Install dependencies  : pass   [required]
[sl-ci-run INFO  234.4 ]: **  Linter                : pass   [optional]
[sl-ci-run INFO  234.4 ]: **  Generate artifacts    : pass   [required]
[sl-ci-run INFO  234.4 ]: **  Run tests             : pass   [required]
[sl-ci-run INFO  234.4 ]: **  Coverage report       : pass   [optional]
[sl-ci-run INFO  234.4 ]: *************** BUILD FRAMEWORK REPORT CARD ****************
```

Connect to https://github.com/strongloop-internal/scrum-loopback/issues/891

### CODE REVIEW

**Comment by Amir-61** [comment]
@slnode test please

File: test/manipulation.test.js [hunk 1]
```diff
@@ -800,8 +800,8 @@ describe('manipulation', function() {
               p.id.should.equal(created.id);
               p.should.not.have.property('_id');
               p.title.should.equal('b');
-              p.should.not.have.property(p.content);
-              p.should.not.have.property(p.comments);
+              should.equal(p.content, undefined);
```
Comment (bajtos) [inline test/manipulation.test.js]: In my experience, this code provides assertion (test failure) messages that are not very helpful:

```js
should.equal(p.content, 'undefined');
```

We usually use the following code instead:

```js
p.should.have.property('content', undefined);
```

Is there any particular reason against using this approach in this pull request?

**Comment by bajtos** [comment]
@Amir-61 the change looks good in principle. I am little bit worried about the assertion code used in unit-tests, see my comment above.

**Comment by Amir-61** [comment]
It seems `loopback-connector-mysql` repo is [broken](http://ci.strongloop.com/job/loopback-connector-mysql/label=amazon-0.10/1252/console) and one test fails which does not seem relevant to this PR; the test which fails is `mysql updateOrCreate should allow updating the instance property value with arrays of nested objects`.

@bajtos I applied your feedback could you please take a look. Thanks!

**Comment by bajtos** [comment]
> It seems loopback-connector-mysql repo is broken and one test fails which does not seem relevant to this PR; the test which fails is mysql updateOrCreate should allow updating the instance property value with arrays of nested objects.

I'll trust your judgement on this.

LGTM, :shipit:

**Comment by bajtos** [comment]
@Amir-61 BTW is this something to back-port to 2.x too?

**Comment by rmg** [comment]
@slnode test please

**Comment by Amir-61** [comment]
@slnode test please

**Comment by Amir-61** [comment]
@slnode test please

**Pull request merged**